Fix Bug in GCNN Parity

chrisrothUT opened the pull request:
I just noticed that parity=True was not working correctly for the GCNN. I fixed the bug and added tests (decided to replace the use_bias test since that is already tested with DenseEquivariant)

PhilipVinc commented on test/models/test_rbm.py:
```diff
@@ -56,11 +56,11 @@ def test_RBMSymm(use_hidden_bias, use_visible_bias, symmetries):
     vmc.advance(1)
 
 
-@pytest.mark.parametrize("use_bias", [True, False])
```
can't we test both partiy and bias?

chrisrothUT replied:
Of course, I was just worried there was too many tests. Also, the bias gets tested in DenseEquivariant so it felt the most redundant. I can change it back of course.

PhilipVinc replied:
Ha, no, then it's ok

PhilipVinc commented:
you should also fix the formatting in order to merge

chrisrothUT commented:
Okay, I made a few changes to make sure the parameters of the different models had the same names and shapes and added a test to make sure that their outputs are the same. This involved adding a transpose to DenseSymmMatrix, which didn't affect runtime at all. 

I also added information to the DenseSymm docstring.
